core/store/src/trie/iterator.rs

```diff
@@ -197,22 +208,25 @@ impl<'a> TrieIterator<'a> {
         prefix
     }
 
-    /// Returns hashes of nodes with paths in [path_begin, path_end). Used by state parts
+    /// Visits all nodes belonging to the interval [path_begin, path_end) in depth-first search order.
+    /// Returns vector of pairs with node hash and node key which is not None if node stores a value.
+    /// Used to generate and apply state parts for state sync.
     pub(crate) fn visit_nodes_interval(
         &mut self,
         path_begin: &[u8],
         path_end: &[u8],
-    ) -> Result<Vec<CryptoHash>, StorageError> {
+    ) -> Result<Vec<(CryptoHash, Option<Vec<u8>>)>, StorageError> {
```

Comment:
I think it is better to introduce a struct here to have more clarity on what the return value is